tfjs-backend-webgpu/src/kernels/Draw.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use backend file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import {KernelConfig, KernelFunc, TensorInfo} from '@tensorflow/tfjs-core';
+import {Draw, DrawAttrs, DrawInputs,} from '@tensorflow/tfjs-core';
+
+import {WebGPUBackend} from '../backend_webgpu';
+import {DrawProgram} from '../draw_webgpu';
+
+export function draw(
+    args: {inputs: DrawInputs, backend: WebGPUBackend, attrs: DrawAttrs}):
+    TensorInfo {
+  const {inputs, backend, attrs} = args;
+  const {image} = inputs;
+  const {canvas, options} = attrs;
+  const [height, width] = image.shape.slice(0, 2);
+  const {imageOptions} = options || {};
+  const alpha = imageOptions ?.alpha || 1;
+
+  //  'rgba8unorm' should work on macOS according to
+  //  https://bugs.chromium.org/p/chromium/issues/detail?id=1298618. But
+  //  failed on macOS/M2. So use 'bgra8unorm' first when available.
+  const format = backend.device.features.has('bgra8unorm-storage') ?
+      'bgra8unorm' :
+      'rgba8unorm';
+  const outShape = [height, width];
+  const program = new DrawProgram(outShape, image.dtype, format);
+  canvas.width = width;
+  canvas.height = height;
+  const backendName = 'webgpu';
+  let gpuContext = canvas.getContext(backendName);
+  let canvasWebGPU;
+  if (!gpuContext) {
+    canvasWebGPU = new OffscreenCanvas(width, height);
```

Comment:
Sorry, forget one. Maybe we should cache the `canvasWebGPU` if needed. Otherwise, it will be created every time when we call the `Draw` API. And the `configure` only needs to be called once if the canvas doesn't change.

Reply:
canvasWebGPU is a fallback when a 2d canvas is used in webgpu context. To support the cache, we should use width, height as key, and I don't know when to free this cache, may be need a new API to free. 
So I suggest donot support cache in this PR, maybe in next PR.